sht3x/sht3x.c

```diff
@@ -92,38 +106,157 @@ int16_t sht3x_read(int32_t* temperature, int32_t* humidity) {
-                                          SHT3X_CMD_READ_STATUS_REG,
+    return sensirion_i2c_delayed_read_cmd(addr, SHT3X_CMD_READ_STATUS_REG,
                                           SHT3X_CMD_DURATION_USEC, &status, 1);
 }
 
+int16_t sht3x_get_status(sht3x_i2c_addr_t addr, uint16_t* status) {
+    return sensirion_i2c_delayed_read_cmd(addr, SHT3X_CMD_READ_STATUS_REG,
+                                          SHT3X_CMD_DURATION_USEC, status, 1);
+}
+
+int16_t sht3x_clear_status(sht3x_i2c_addr_t addr) {
+    return sensirion_i2c_write_cmd(addr, SHT3X_CMD_CLR_STATUS_REG);
+}
+
 void sht3x_enable_low_power_mode(uint8_t enable_low_power_mode) {
     sht3x_cmd_measure =
         enable_low_power_mode ? SHT3X_CMD_MEASURE_LPM : SHT3X_CMD_MEASURE_HPM;
 }
 
-int16_t sht3x_read_serial(uint32_t* serial) {
+void sht3x_set_power_mode(sht3x_measurement_mode_t mode) {
+
+    switch (mode) {
+        case SHT3X_MEAS_MODE_LPM: {
+            sht3x_cmd_measure = SHT3X_CMD_MEASURE_LPM;
+            break;
+        }
+        case SHT3X_MEAS_MODE_MPM: {
+            sht3x_cmd_measure = SHT3X_CMD_MEASURE_MPM;
+            break;
+        }
+        case SHT3X_MEAS_MODE_HPM: {
+            sht3x_cmd_measure = SHT3X_CMD_MEASURE_HPM;
+            break;
+        }
+        default: {
+            sht3x_cmd_measure = SHT3X_CMD_MEASURE_HPM;
+            break;
+        }
+    }
+    return;
+}
+
+int16_t sht3x_read_serial(sht3x_i2c_addr_t addr, uint32_t* serial) {
     int16_t ret;
     uint8_t serial_bytes[4];
 
-    ret = sensirion_i2c_write_cmd(SHT3X_ADDRESS, SHT3X_CMD_READ_SERIAL_ID);
-    if (ret)
-        return ret;
-
+    ret = sensirion_i2c_write_cmd(addr, SHT3X_CMD_READ_SERIAL_ID);
     sensirion_sleep_usec(SHT3X_CMD_DURATION_USEC);
 
-    ret = sensirion_i2c_read_words_as_bytes(SHT3X_ADDRESS, serial_bytes,
-                                            SENSIRION_NUM_WORDS(serial_bytes));
-    *serial = sensirion_bytes_to_uint32_t(serial_bytes);
+    if (ret == STATUS_OK) {
+
+        ret = sensirion_i2c_read_words_as_bytes(
+            addr, serial_bytes, SENSIRION_NUM_WORDS(serial_bytes));
+        *serial = sensirion_bytes_to_uint32_t(serial_bytes);
+    }
     return ret;
 }
 
 const char* sht3x_get_driver_version(void) {
     return SHT_DRV_VERSION_STR;
 }
 
-uint8_t sht3x_get_configured_address(void) {
-    return SHT3X_ADDRESS;
+int16_t sht3x_set_alert_thd(sht3x_i2c_addr_t addr, sht3x_alert_thd_t thd,
+                            uint16_t humidity, int16_t temperature) {
+    int16_t ret;
+    uint32_t tmp;
+    uint16_t limitVal = 0U;
+
+    /* convert inputs to alert threshold word */
+    tmp = humidity * 65535U;
+    tmp = tmp / 1000U; /*humidity was provided in 10%RH*/
+    limitVal = ((uint16_t)tmp & SHT3X_HUMIDITY_LIMIT_MSK);
+
+    tmp = (uint32_t)(temperature + 450);
+    tmp = tmp * 65535U;
+    tmp = tmp / 1750U;
+    tmp = (tmp >> 7);
+    limitVal |= ((uint16_t)tmp & SHT3X_TEMPERATURE_LIMIT_MSK);
```

Comment:
I would prefer to use the same  temperature and humidity format as for measurements (value * 1000).

This way we can use a general conversion function from ticks to temperature/humidity and vice-versa.

I suggest something like this
```
void tick_to_temperature(uint16_t tick, int32_t* temperature) {
    *temperature = ((21875 * (int32_t)tick) >> 13) - 45000;
}

void tick_to_humidity(uint16_t tick, int32_t* humidity) {
    *humidity = ((12500 * (int32_t)tick) >> 13);
}

void temperature_to_tick(int32_t temperature, uint16_t* tick) {
   *tick = (uint16_t)((temperature * 12271 + 552195000) >> 15);
}

void humidity_to_tick(int32_t humidity, uint16_t* tick) {
    *tick = (uint16_t)((humidity * 21474) >> 15);
}
```

Then we get the `limitVal` as follows:
```c
uint16_t t_raw, rh_raw, limit_value = 0U;

temperature_to_tick(temperature, &t_raw);
humidity_to_tick(humidity, &rh_raw);

limit_value = (rh_raw & SHT3X_HUMIDITY_LIMIT_MSK);
limit_value |= ((t_raw >> 7) & SHT3X_TEMPERATURE_LIMIT_MSK);
```

Reply:
Agreed, it will be better aligned. But it is a bit less memory efficient though (4 extra bits required as parameters that will be truncated in the set_alert_thd).